features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/PrivacyOverrideManager.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal
+
+import android.view.View
+import com.datadog.android.sessionreplay.ImagePrivacy
+import com.datadog.android.sessionreplay.PrivacyLevel
+import com.datadog.android.sessionreplay.TextAndInputPrivacy
+import com.datadog.android.sessionreplay.TouchPrivacy
+import java.lang.ref.WeakReference
+import java.util.concurrent.ConcurrentHashMap
+
+internal object PrivacyOverrideManager {
+    private val overridesMap = ConcurrentHashMap<WeakViewKey, PrivacyOverride>()
```

Comment:
Why not using directly a `WeakHashMap` ? The key is already `weak` in there: https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/WeakHashMap.html